sourcegraph/learn
# Bump to Next.js 11

### What should this PR do?
Resolves [DEVED-115](https://sourcegraph.atlassian.net/browse/DEVED-115) by moving us to Next.js 11.

It does not implement a complete `img` overhaul, for the following reason: the `Image` component does not play well with `next export`, so it seems like we will need to circle back to this once we have decided what we are doing about our images more generally.  We have the option to configure a custom [`loader`](https://nextjs.org/docs/api-reference/next/image#loader), which would enable us to use an image optimization provider (like Cloudinary), or to write a custom loader that interfaces with a CDN. Here's a [thread](https://www.reddit.com/r/nextjs/comments/omywo5/is_there_any_point_of_using_next_image_when_the/) that lays out some of the stakes.

From the discord: "[...] You can still use next/image with next export and choose an external image optimization provider
You just can't use the default Next.js one, because you're opting not to use a server with export. [...] The next release of Next.js is improving this experience, with a static loader and better docs around using next/image with next export.
And, just to reiterate, build time image optimization is one of the most requested features we have, and we are working on an RFC for our proposed solution!" <-- this last remark dated 7/21/2021

Another option might be using [`next-optimized-images`](https://github.com/cyrilwanner/next-optimized-images), as described [here](https://maxrohde.com/2021/07/25/next-js-11-images-with-static-export/), for example.

Either way, let's definitely come back to it.

### Why are we making this change?
- We want the project to stay current with Next.js

### What are the acceptance criteria? 
- Everything should work as it currently does in production.

### How should this PR be tested?
- Check out the branch, remove `node_modules`, and run `npm ci` to install dependencies.
- Verify the above.

## Pull request process

**Reviewers**:

1. Test functionality using the criteria above.
2. Offer tips for efficiency, feedback on best practices, and possible alternative approaches and things that may not have been considered.
3. For shorter, "quick" PRs, use your best judgement on #​2.
4. Use a collaborative approach and provide resources and/or context where appropriate.
5. Provide screenshots/grabs where appropriate to show findings during review.

**Reviewees**:

1. Prefer incremental and appropriately-scoped changes.
2. Leave a comment on things you want explicit feedback on.
3. Respond clearly to comments and questions.
